### PR TITLE
Add ScheduleGrid

### PR DESCRIPTION
**Pull request by danielchen-pyc**
# Changes
- Added schedule grid and addCourse function for the timetable. 
- @johnagl 

# Related Issues
- #71

### CODE REVIEW

File: client/src/components/ScheduleGrid.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        }, );
+        half++;
+    }
+    return rows;
+};
+
+
+function addCourse(rows: any, day: any, startTime: number, endTime: number, courseName: string) {
+    if (startTime >= endTime || courseName === "" || courseName === null) return;
+    let colnum = day + 1;
```
Comment (johnagl) [inline client/src/components/ScheduleGrid.tsx:59]: This variable is currently greyed out, but everything looks functional. Is `colnum` necessary anywhere?

File: client/src/components/ScheduleGrid.tsx [hunk 2]
```diff
@@ -57,14 +57,20 @@ let makerows = function (start: number, end: number) {
 function addCourse(rows: any, day: any, startTime: number, endTime: number, courseName: string) {
     if (startTime >= endTime || courseName === "" || courseName === null) return;
     let colnum = day + 1;
+<<<<<<< HEAD
     for (let row of rows) {
```
Comment (johnagl) [inline client/src/components/ScheduleGrid.tsx]: Looks like these merge conflicts werent fixed properly indicated by the `<<<<<< HEAD` ...

**Comment by johnagl** [comment]
@danielchen-pyc Fix the last merge conflict and this should be good to merge!

**Comment by danielchen-pyc** [comment]
@johnagl Fixed the merge conflict! Please have a look. Thank you!

**Review by johnagl — APPROVED** [review]
LGTM :)

**Pull request merged**